compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12882,6 +12882,88 @@ TR::Node *ificmpneSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier
    return node;
    }
 
+/**
+ * \brief
+ * Simplify add/subtract operations under integer comparison nodes
+ * to save an add/sub instruction during codegen.
+ *
+ * \details
+ *
+ * e.g. This following is a before-after comparison.
+ *
+ * Before:
+ * ificmple --> block_17 BBStart at n13n ()
+ *   iadd (X>=0 cannotOverflow )
+ *     ==>iRegLoad (in GPR_0112) (cannotOverflow canBeAssignedToHPR SeenRealReference )
+ *     iconst -1 (X!=0 X<=0 )
+ *   ==>iconst 0 (X==0 X>=0 X<=0 )
+ *
+ *
+ * After:
+ * ificmple --> block_17 BBStart at n13n ()
+ *   ==>iRegLoad (in GPR_0112) (cannotOverflow canBeAssignedToHPR SeenRealReference )
+ *   iconst 1 (X==0 X>=0 X<=0 )
+ *
+ *
+ * The iadd node needs to be single reference and marked as 'cannotOverflow' to
+ * ensure functional correctness.
```

Comment:
I'm currently working on signed/unsigned long, int, byte, short, and etc. I'll test and update this MR.